qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -400,13 +400,19 @@ def creation_date(self):
         """Return creation date."""
         return self._creation_date
 
-    def job_id(self):
-        """Return backend determined id.
+    def job_id(self, timeout=60):
+        """Return the job ID assigned by the API.
 
-        If the Id is not set because the job is already initializing, this call
-        will block until we have an Id.
+        If the job ID is not set because the job is already initializing, this
+        call will block until a job ID is available of the timeout is reached.
+
+        Args:
+            timeout (float): number of seconds to wait for the job id.
```

Comment:
super nitpicking... can you change `id` to `ID` so they're all consistent in this method?